src/io/buf_read/fill_buf.rs

```diff
@@ -0,0 +1,32 @@
+use std::pin::Pin;
+
+use futures_io::AsyncBufRead;
+
+use crate::future::Future;
+use crate::io;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+pub struct FillBufFuture<'a, R: ?Sized> {
+    reader: &'a mut R,
+}
+
+impl<'a, R: ?Sized> From<&'a mut R> for FillBufFuture<'a, R> {
+    fn from(reader: &'a mut R) -> Self {
+        Self { reader }
+    }
+}
```

Comment:
Given there is only a single creator of this future, is this `From` impl necessary? If it is, could we perhaps add a comment to explain why?

Reply:
I wanted to make `reader` field private, but yeah, it could be an associated function, or just make `reader` `pub(crate)`.

Reply:
@tirr-c This is not a big issue, but given a choice I'd prefer a `pub(crate)` field and a function in order to not leak any unintended APIs publicly. This `impl` is currently public in the sense that the user could do `.into()` to convert a buffer into the future.